src/Raven.Server/Documents/Indexes/MapReduce/Static/MapReduceIndex.cs

```diff
@@ -778,9 +780,20 @@ public void Dispose()
 
                 private static void ThrowMissingGroupByFieldsInMapOutput(object output, Dictionary<string, CompiledIndexField> groupByFields, AbstractStaticIndexBase compiledIndex)
                 {
+                    string outputString;
+                    if (output is ObjectInstance oi)
```

Comment:
I've noticed in a lot of places ObjectInstance is swapped to JsObject, but not here. Double checking if this is correct?

Reply:
I made this generic in purpose, so it could also print `Proxy`, `Number`, `Date` etc also. This addition was made as found the error message quite lacking when I was migrating.

RavenDB seemed to check explicit type of `ObjectInstance` so that is basically anything manually built as object or anything in script declared via curlies (`const object = { foo: "bar" };`. Now it's the same as having `JsObject` which is sealed and cannot be something like `Date` or `ArrayPrototype` I kept that behavior elsewhere not to break expectations.